### PR TITLE
Allow users to set up profiles that are shared with future matches

### PR DESCRIPTION
**Pull request by felixtran39**
**Description**
This is the implementation for allowing users to set profiles for themselves. In this update, a welcome card is sent to each user individually (regardless of whether they join before or after Icebreaker is added). This welcome card includes a "Set up profile" button which can be used to configure their profile. Once saved, this profile will be shared with future matches in the match card. The match card also includes a "Update profile" button, which can be used to conveniently update their profile for future matches.

**User facing changes**
Users will always receive a personal welcome card from Icebreaker with an option to set up their profile. Match cards include an option to update their profile. Match cards share a user's profile with their match if it exists.

**Follow-up work**
Currently, users are not stored in the database until a user chooses to "Pause all matches" (users using Icebreaker may never be stored in the database). This can lead to issues when setting up a profile before the user is in the database. Another feature currently in development addresses this issue by storing all users that are part of a team with Icebreaker no matter what.

Tests can be configured that cover different lengths of profiles for users (represented as strings), the types of characters supported, etc.

### CODE REVIEW

File: Source/Icebreaker/Bot/IcebreakerBot.cs
Comment (beauzeaux) [inline Source/Icebreaker/Bot/IcebreakerBot.cs:236]: Can you comment on why we are changing this to use the Bot Framework id instead of the AAD Id?

File: Source/Icebreaker/Bot/IcebreakerBot.cs
Comment (beauzeaux) [inline Source/Icebreaker/Bot/IcebreakerBot.cs:250]: If we are changing this to the bot framework id this shouldn't be marked as "aad" id. These are two separate systems with two different identifiers for the user.

File: Source/Icebreaker/Bot/IcebreakerBot.cs
Comment (beauzeaux) [inline Source/Icebreaker/Bot/IcebreakerBot.cs:338]: Please rename to something denoting it is the 'action' such as `cardActionType` or `cardAction`

File: Source/Icebreaker/Bot/IcebreakerBot.cs
Comment (beauzeaux) [inline Source/Icebreaker/Bot/IcebreakerBot.cs:360]: Please add a default case with a warning logged on what type of action was attempted.

**Review by beauzeaux — CHANGES_REQUESTED** [review]
Good start, but please review the change from 'aadId' to 'id'. These are not interchangeable.

File: Source/Icebreaker/Bot/IcebreakerBot.cs
Comment (felixtran39) [reply Source/Icebreaker/Bot/IcebreakerBot.cs:236]: When attempting to access a ChannelAccount's AadObjectId in MatchingService.cs, it always returns null (might be a bug with ChannelAccount). In order to reference the user's profiles during matching the Id had to be used to avoid this issue.

File: Source/Icebreaker/Bot/IcebreakerBot.cs
Comment (beauzeaux) [inline Source/Icebreaker/Bot/IcebreakerBot.cs:291]: Same as before, please don't label this an AAD Id.

**Review by beauzeaux — COMMENTED** [review]
I'll wait on @aosolis to approve as well but LGTM.

**Comment by felixtran39** [comment]
@beauzeaux I was able to fix the issue preventing me from using the AadId, so the changes regarding AadId->Id should be back to the original.

File: Source/Icebreaker/Bot/IcebreakerBot.cs
Comment (felixtran39) [reply Source/Icebreaker/Bot/IcebreakerBot.cs]: No longer making this change

File: Source/Icebreaker/Bot/IcebreakerBot.cs
Comment (aosolis) [inline Source/Icebreaker/Bot/IcebreakerBot.cs:360]: scrub for hardcoded strings and move to resources

File: Source/Icebreaker/Bot/IcebreakerBot.cs
Comment (aosolis) [inline Source/Icebreaker/Bot/IcebreakerBot.cs:350]: nit: remove blank line here, add blank lines before line 374 (switch) and line 366 (default case)

File: Source/Icebreaker/Helpers/AdaptiveCards/PairUpNotificationAdaptiveCard.json
Comment (aosolis) [inline Source/Icebreaker/Helpers/AdaptiveCards/PairUpNotificationAdaptiveCard.json:82]: Move to resources

File: Source/Icebreaker/Services/MatchingService.cs
Comment (aosolis) [inline Source/Icebreaker/Services/MatchingService.cs:82]: Can we merge these 2 DB calls? Something like GetAllUsers?

File: Source/Icebreaker/Services/MatchingService.cs
Comment (aosolis) [inline Source/Icebreaker/Services/MatchingService.cs:158]: The parameter grouping here feels off. `Tuple<TeamsChannelAccount, string> person1, Tuple<TeamsChannelAccount, string> person2` seems more logical

File: Source/Icebreaker/Services/MatchingService.cs
Comment (felixtran39) [reply Source/Icebreaker/Services/MatchingService.cs:82]: Are you suggesting that `GetAllUsers` will return `Dictionary<string, Tuple<string, string>>`? Based on how some of the other helper functions are currently written I would have to break it apart into two dictionaries at the end.

**Comment by felixtran39** [comment]
Replaced wwith #131

**Pull request closed without merging**
